### ISSUE TITLE
The game keeps crashing if I try to use the level I made, please help...

### ISSUE THREAD

**Issue by KeoughDarklance** [ProgSys/pg_disatools#29]
I can't figure out what could be wrong, could anyone help?

[The Grindstone.zip](https://github.com/user-attachments/files/18018598/The.Grindstone.zip)

**Comment by ProgSys** [comment]
Your zip file got somehow corrupted! Do you know what you did to cause this? 
I fixed the zip file and it now loads fine for me: 
[https://www.dropbox.com/scl/fi/7ae7mtvyfcmrn7glxnhl9/TheGrindstone.zip?rlkey=86sxudsx4rbyj9gl1591ljamj&dl=0](https://www.dropbox.com/scl/fi/7ae7mtvyfcmrn7glxnhl9/TheGrindstone.zip?rlkey=86sxudsx4rbyj9gl1591ljamj&dl=0)

**Comment by KeoughDarklance** [comment]
Thank you getting back to me so quickly. Unfortunately that did not fix the issue. Loading it wasn't a problem, it's that after a while it crashes. I will list the steps below. As for how it got corrupted I am unsure. It could be related to how the tool kept crashing every minute or two after I started adding geo panels and actors so I had to save after every little thing I did or loose progress.

Steps:
I spawn in a scout and activate the blue geo panel to teleport him to the ring.
I spawn in a brawler and remove her items and shoot her with the scout.
I spawn in the archer I planned to level up and move her to the blue panel to teleport to the ring.
I have the scout throw her into the island inside the ring.
I shoot the brawler again and move the scout to the blue tile to teleport back.
I shoot the brawler 1 final time then bring out my healer to remove sleep from the brawler, then I move the brawler to the cyan panel and have her throw the clone geo onto the cyan panel next to her.
It is here that the game usually crashes although sometimes I get as far as the third clone spawning and attacking it before crashing.

**Comment by ProgSys** [comment]
That my tool crashes is definitely something I can fix. Maybe you also have a some steps for that too?
That the game crashes is a different topic, as I only modified the initial loading of the map. After that, the original game code is run.
But I will have a look, maybe I can find something.

**Comment by ProgSys** [comment]
I sadly don't have the exact units you have, but for me your map works fine. No crash here.
Can you try something out for me? Can you set the game to window mode instead of full screen, then when you spawned into your map, press F2. A terminal should open. Inside simply type "u" and press Enter. This will unload all custom code. Can you see if the game still crashes after that?

**Comment by ProgSys** [comment]
I think I found the problem. The game cant handle Geo Symbols that are not alliance "Enemy, third party". Otherwise it tries tries to use out of bounds function slots for some reason.

[I released a new version (v1.0.1)!](https://github.com/ProgSys/pg_disatools) 
It now has some smaller fixes. **Plus** if you open you map, do some change and resave it, then it will set alliance to the correct values.

If you find a repo on how to crash the map editor, I would appreciate it if you share!
